sql/core_clients_last_seen_v1.sql

```diff
@@ -0,0 +1,49 @@
+WITH
+  current_sample AS (
+  SELECT
+    -- Include a dummy last_date struct to make schema match with `previous`.
+    STRUCT(DATE('2000-01-01'), DATE('2000-01-01')) AS last_date,
+    * EXCEPT (submission_date,
+      generated_time)
+  FROM
+    core_clients_daily_v1
+  WHERE
+    submission_date = @submission_date ),
+  previous AS (
+  SELECT
+    * EXCEPT (submission_date,
+      generated_time)
+  FROM
+    core_clients_last_seen_v1
+  WHERE
+    submission_date = DATE_SUB(@submission_date, INTERVAL 1 DAY)
+    AND last_date.seen > DATE_SUB(@submission_date, INTERVAL 28 DAY) )
+SELECT
+  @submission_date AS submission_date,
+  CURRENT_DATETIME() AS generated_time,
+  -- This last_date struct is used to record the last date that a particular
+  -- client met various usage criteria.
+  STRUCT (
+    -- last_date.seen is the last date we recieved any core ping at all.
+    IF(current_sample.client_id IS NOT NULL,
+      @submission_date,
+      previous.last_date.seen) AS seen,
```

Comment:
this is most excellent. but I think we should probably enable clustering on this value (thought maybe not immediately), in which case it can't be in a `struct` 😭 . thoughts?

Reply:
The desire for clustering here would be to make calculations on smaller windows more efficient?

Reply:
it would optimize `WHERE ... last_date.seen > DATE_SUB(@submission_date, INTERVAL 28 DAY)`

Reply:
Got rid of the struct and made these both top-level fields.